File: Hotels/Models/Room.cs
using System.ComponentModel.DataAnnotations;

namespace Hotels.Models
{
    public class Room
    {
        public int Id { get; set; }

        [Required]
        public string Name { get; set; }

        public bool IsAvailable { get; set; } = true;

        [Required]
        public virtual RoomType RoomType { get; set; }
    }
}

File: Hotels/Scripts/Room.js
function Room(object) {
    if (object) {
        this.id = object.id;
        this.name = object.name;
        this.isAvailable = object.isAvailable;
        this.roomTypeId = object.roomTypeId;
    } else {
        this.id = null;
        this.name = null;
        this.isAvailable = null;
        this.roomTypeId = null;
    }
}

Room.prototype.toString = function () {
    return this.name + " - " + this.roomTypeId;
}